python/tests/estimators/test_keras_estimators.py

```diff
@@ -79,40 +89,63 @@ def _get_estimator(self, model, optimizer='adam', loss='categorical_crossentropy
-                                       kerasLoss=loss,
-                                       kerasFitParams=keras_fit_params,
+                                       kerasOptimizer='adam',
+                                       kerasLoss='categorical_crossentropy',
                                        modelFile=model_filename)
         return estm
 
     def setUp(self):
         self.temp_dir = tempfile.mkdtemp()
         self.input_col = 'kerasTestImageUri'
-        self.label_col = 'kerasTestlabel'
+        self.label_col = 'kerasTestLabel'
+        self.one_hot_col = 'kerasTestOneHot'
         self.output_col = 'kerasTestPreds'
 
     def tearDown(self):
         shutil.rmtree(self.temp_dir, ignore_errors=True)
 
-    def test_valid_workflow(self):
+    def test_single_training(self):
         # Create image URI dataframe
         label_cardinality = 10
         image_uri_df = self._create_train_image_uris_and_labels(
             repeat_factor=3, cardinality=label_cardinality)
 
-        # We need a small model so that machines with limited resources can run it
-        model = Sequential()
-        model.add(Flatten(input_shape=(299, 299, 3)))
-        model.add(Dense(label_cardinality))
-        model.add(Activation("softmax"))
-
+        model = self._get_model(label_cardinality)
         estimator = self._get_estimator(model)
+        estimator.setKerasFitParams({'verbose': 1})
         self.assertTrue(estimator._validateParams())
-        transformers = estimator.fit(image_uri_df)
-        self.assertEqual(1, len(transformers))
-        self.assertIsInstance(transformers[0]['transformer'], KerasImageFileTransformer)
+
+        transformer = estimator.fit(image_uri_df)
+        self.assertIsInstance(transformer, KerasImageFileTransformer, "output should be KIFT")
+        output_df = transformer.transform(image_uri_df)
+        self.assertEqual(len(output_df.select(F.col(self.output_col)).collect()), len(image_uri_df.collect()),
+                         "output column should exist and be equal sized")
+
+    def test_tuning(self):
+        # Create image URI dataframe
+        label_cardinality = 2
+        image_uri_df = self._create_train_image_uris_and_labels(
+            repeat_factor=3, cardinality=label_cardinality)
+
+        model = self._get_model(label_cardinality)
+        estimator = self._get_estimator(model)
+
+        paramGrid = (
+            ParamGridBuilder()
+            .addGrid(estimator.kerasFitParams, [{"batch_size": 32}, {"batch_size": 64}])
+            .build()
+        )
+
+        bc = BinaryClassificationEvaluator(rawPredictionCol=self.output_col, labelCol=self.label_col)
+        cv = CrossValidator(estimator=estimator, estimatorParamMaps=paramGrid, evaluator=bc, numFolds=2)
+
+        transformer = cv.fit(image_uri_df)
+        self.assertIsInstance(transformer.bestModel, KerasImageFileTransformer, "best model should be KIFT")
+        output_df = transformer.transform(image_uri_df)
```

Comment:
Here I think we can rely on the KIFT tests to confirm that `transform` produces the right result given a well-configured transformer. Maybe instead we should validate the params of the KIFT are set correctly (e.g. that it has the correct input & output columns?)

Reply:
we should validate the params of the KIFT are set correctly (e.g. that it has the correct input & output columns?)